### ISSUE TITLE
IDE0048 triggers for `is null` but not for `is not null` in null-coalescing expressions

### ISSUE THREAD

**Issue by rampaa** [dotnet/roslyn#81488]
**Version Used**: 
Visual Studio 2026 v18.0.0
.NET 10

**Steps to Reproduce**:

1. Create a class like the following:
```csharp
internal sealed class Test
{
    public override bool Equals(object? obj)
    {
        return Equals(obj as Test);
    }

    public override int GetHashCode()
    {
        return 0;
    }

    public static bool operator ==(Test? left, Test? right) => left?.Equals(right) ?? right is null; // <- Triggers IDE0048

    public static bool operator !=(Test? left, Test? right) => !left?.Equals(right) ?? right is not null; // <- Does not trigger IDE0048
}
```
2. Observe that `left?.Equals(right) ?? right is null;` triggers IDE0048, while `!left?.Equals(right) ?? right is not null;` does not.

**Diagnostic Id**:
IDE0048 Parentheses should be added for clarity

**Expected Behavior**:
IDE0048 should behave consistently, it should either trigger for both cases or for neither. For reference, Rider considers parentheses redundant in both cases.

**Actual Behavior**:
IDE0048 is triggered for the `is null` check but not for the `is not null` check.
